Source/SwiftLintBuiltInRules/Rules/Idiomatic/OneDelarationPerFileRule.swift

```diff
@@ -0,0 +1,118 @@
+    var configuration = SeverityConfiguration<Self>(.error)
+
+    static let description = RuleDescription(
+        identifier: "one_declaration_per_file",
+        name: "One Declaration Per File",
+        description: "One declaration per file is allowed, extensions are an exception",
+        kind: .idiomatic,
+        nonTriggeringExamples: [
+            Example("""
+                    class Car {
+                        var make: String
+                        var model: String
+                        init(make: String, model: String) {
+                            self.make = make
+                            self.model = model
+                        }
+                    }
+                    """),
+            Example("""
+                    class Car {
+                        var make: String
+                        var model: String
+                        init(make: String, model: String) {
+                            self.make = make
+                            self.model = model
+                        }
+                    }
+                    extension Car {
+                        func drive() {
+                        }
+                        func stop() {
+                        }
+                    }
+                    """)
+        ],
+        triggeringExamples: [
+            Example("""
+                    class Car {
+                        var make: String
+                        var model: String
+                        init(make: String, model: String) {
+                            self.make = make
+                            self.model = model
+                        }
+                    }
+                    class Bike {
+                        func ride() {
+                        }
+                        func stop() {
+                        }
+                    }
+                    """),
+            Example("""
+                    protocol Identifiable {
+                        var identifier: String { get }
+                    }
+                    enum IdentifiableTypes: String {
+                        case linear, composite
+                    }
+                    """),
+            Example("""
+                    struct BasicProfile {
+                        var id: Int
+                        var name: String
+                    }
+                    struct DetailedProfile {
+                        var basic: BasicProfile
+                        var age: String
+                        var genderRaw: String
+                    }
+                    """)
+        ]
+    )
+}
+
+private extension OneDelarationPerFileRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        var declarationsCount: Int = 0
+        override var skippableDeclarations: [any DeclSyntaxProtocol.Type] {
+            return .allExcept(ClassDeclSyntax.self, StructDeclSyntax.self, EnumDeclSyntax.self, ProtocolDeclSyntax.self)
+        }
```

Comment:
You can skip everything here. This is about the children of declarations. The declarations themselves will always be visited.

```suggestion
        override var skippableDeclarations: [any DeclSyntaxProtocol.Type] { .all }
```